Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM gradle:7.4.2-jdk8 AS builder
+WORKDIR /build
+COPY . .
+
+RUN tr -d '\015' <gradlew >gradlew-unix && \
+    chmod +x gradlew-unix && \
+    ./gradlew-unix jar
+
+
+FROM azul/zulu-openjdk:8u202
+WORKDIR /app
+
+COPY --from=builder /build/grasscutter.jar grasscutter.jar
+
+# mount these folder/file at runtime please
+# resources/ data/ config.json keystore.p12
+COPY Grasscutter-Protos/proto/ ./proto
```

Comment:
Protos are already included and their classes are now automatically built

![image](https://user-images.githubusercontent.com/27217031/166292311-98a2be71-3de4-4933-948f-5efc2d9ad35f.png)

Reply:
done.